corehq/apps/locations/models.py

```diff
@@ -322,7 +322,10 @@ def filter_by_user_input(self, domain, user_input):
         """
         query = None
         for part in user_input.split('/'):
-            query = self.get_queryset_descendants(query) if query is not None else self
+            if query is None:
+                query = self.filter(domain=domain)
+            else:
+                query = self.get_queryset_descendants(query)
```

Comment:
Unless I'm missing something with how `get_queryset_descendants` behaves, isn't it the case that this code path never filters by domain?

Reply:
Oh sorry I should read the PR description closer
> Lower query levels need not be filtered by domain since those levels are constrained by parent_id, where the parent should always be in the same domain as its children.
 
Nevermind!